server/tso/allocator_manager.go

```diff
@@ -288,28 +289,62 @@ func (am *AllocatorManager) campaignAllocatorLeader(parentCtx context.Context, a
 	}
 }
 
-// AllocatorDaemon is used to update every allocator's TSO.
-func (am *AllocatorManager) AllocatorDaemon(serverCtx context.Context) {
+// AllocatorDaemon is used to update every allocator's TSO and check whether we have
+// any new local allocator that needs to be set up.
+func (am *AllocatorManager) AllocatorDaemon(serverCtx context.Context, serverCtxCancel context.CancelFunc) {
 	tsTicker := time.NewTicker(UpdateTimestampStep)
 	defer tsTicker.Stop()
+	checkerTicker := time.NewTicker(checkAllocatorStep)
+	defer checkerTicker.Stop()
 
 	for {
 		select {
 		case <-tsTicker.C:
-			// Filter out allocators without leadership and uninitialized
-			allocatorGroups := am.getAllocatorGroups(FilterUninitialized(), FilterUnavailableLeadership())
-			// Update each allocator concurrently
-			for _, ag := range allocatorGroups {
-				am.wg.Add(1)
-				go am.updateAllocator(ag)
-			}
-			am.wg.Wait()
+			// Update the Local TSO Allocator leaders TSO in memory concurrently.
+			am.allocatorUpdater()
+		case <-checkerTicker.C:
+			// Check if we have any new dc-location configured, if yes,
+			// then set up the corresponding local allocator.
+			am.allocatorChecker(serverCtx, serverCtxCancel)
 		case <-serverCtx.Done():
 			return
 		}
 	}
 }
 
+func (am *AllocatorManager) allocatorChecker(ctx context.Context, ctxCancel context.CancelFunc) {
+	clusterDCLocations, err := am.GetClusterDCLocations()
+	if err != nil {
+		log.Error("check new allocators failed, can't get cluster dc-locations", errs.ZapError(err))
+	}
+	allocatorGroups := am.getAllocatorGroups()
+	for dcLocation := range clusterDCLocations {
+		if slice.NoneOf(allocatorGroups, func(i int) bool {
+			return allocatorGroups[i].dcLocation == dcLocation
+		}) {
+			if err := am.SetUpAllocator(ctx, ctxCancel, dcLocation, election.NewLeadership(
+				am.member.Client(),
+				am.getAllocatorPath(dcLocation),
+				fmt.Sprintf("%s local allocator leader election", dcLocation),
+			)); err != nil {
+				log.Error("check new allocators failed, can't set up a new local allocator", zap.String("dc-location", dcLocation), errs.ZapError(err))
+				continue
+			}
+		}
+	}
+}
+
+func (am *AllocatorManager) allocatorUpdater() {
+	// Filter out allocators without leadership and uninitialized
+	allocatorGroups := am.getAllocatorGroups(FilterUninitialized(), FilterUnavailableLeadership())
+	// Update each allocator concurrently
+	for _, ag := range allocatorGroups {
+		am.wg.Add(1)
+		go am.updateAllocator(ag)
+	}
+	am.wg.Wait()
+}
```

Comment:
I'm wondering why `allocatorUpdater` doesn't introduce the `ctx` and `cancelFunc`.

Reply:
I found the context usage here is so disorganized...I need to sort it out.

Reply:
Thanks for pointing out!

Reply:
Updated. `allocatorUpdater` doesn't need the context because an `allocatorGroup` already has one. `allocatorPatroller` needs context because it needs to set up an allocator which will be added into its `allocatorGroup`.